tools/serverpod_cli/bin/generator/protocol_generator_dart.dart

```diff
@@ -9,215 +11,227 @@ class ProtocolGeneratorDart extends ProtocolGenerator {
+                ..named = true
+                ..type = TypeReference((t) => t
+                  ..symbol = 'ServerpodClientErrorCallback'
+                  ..url = serverPodUrl(false)
+                  ..isNullable = true)),
+              Parameter((p) => p
+                ..name = 'authenticationKeyManager'
+                ..named = true
+                ..type = TypeReference((t) => t
+                  ..symbol = 'AuthenticationKeyManager'
+                  ..url = serverPodUrl(false)
+                  ..isNullable = true)),
+            ])
+            ..initializers.add(refer('super').call([
+              refer('host'),
+              refer('Protocol', 'protocol.dart').call([])
+            ], {
+              'context': refer('context'),
+              'errorHandler': refer('errorHandler'),
+              'authenticationKeyManager': refer('authenticationKeyManager'),
+            }).code);
+        } else {
+          c
+            ..requiredParameters.add(Parameter((p) => p
+              ..type = refer('ServerpodClientShared', serverPodUrl(false))
+              ..name = 'client'))
+            ..initializers.add(refer('super').call([refer('client')]).code);
+        }
+        c.body = Block.of([
+          for (var endpointDef in protocolDefinition.endpoints)
+            refer(endpointDef.name)
+                .assign(refer(_endpointClassName(endpointDef.name))
+                    .call([refer('this')]))
+                .statement,
+          if (hasModules)
+            refer('modules')
+                .assign(refer('_Modules').call([refer('this')]))
+                .statement,
+        ]);
+      }))
+      ..methods.addAll([
+        Method(
+          (m) => m
+            ..name = 'endpointRefLookup'
+            ..annotations.add(refer('override'))
+            ..type = MethodType.getter
+            ..returns = TypeReference((t) => t
+              ..symbol = 'Map'
+              ..types.addAll([
+                refer('String'),
+                refer('EndpointRef', serverPodUrl(false)),
+              ]))
+            ..body = literalMap({
+              for (var endpointDef in protocolDefinition.endpoints)
+                '$modulePrefix${endpointDef.name}': refer(endpointDef.name)
+            }).code,
+        ),
+        if (config.type == PackageType.server)
+          Method(
+            (m) => m
+              ..name = 'moduleLookup'
+              ..annotations.add(refer('override'))
+              ..type = MethodType.getter
+              ..returns = TypeReference((t) => t
+                ..symbol = 'Map'
+                ..types.addAll([
+                  refer('String'),
+                  refer('ModuleEndpointCaller', serverPodUrl(false)),
+                ]))
+              ..body = literalMap({
+                for (var module in config.modules)
+                  module.nickname: refer('modules').property(module.nickname),
+              }).code,
+          ),
+      ])));
```

Comment:
Trailing commas for readability.

Reply:
Like this? 6019dd7

Reply:
Exactly, makes it easier to find the matching parenthesis when working with the code.